lua/telescope/actions/set.lua

```diff
@@ -113,19 +113,18 @@ action_set.edit = function(prompt_bufnr, command)
   if entry_bufnr then
     edit_buffer(command, entry_bufnr)
   else
-
     -- check if we didn't pick a different buffer
     -- prevents restarting lsp server
     if vim.api.nvim_buf_get_name(0) ~= filename or command ~= "edit" then
       filename = path.normalize(vim.fn.fnameescape(filename), vim.loop.cwd())
       vim.cmd(string.format("%s %s", command, filename))
     end
+  end
 
-    if row and col then
-      local ok, err_msg = pcall(a.nvim_win_set_cursor, 0, {row, col})
-      if not ok then
-        log.debug("Failed to move to cursor:", err_msg, row, col)
-      end
+  if row and col then
+    local ok, err_msg = pcall(a.nvim_win_set_cursor, 0, {row, col})
+    if not ok then
+      log.debug("Failed to move to cursor:", err_msg, row, col)
```

Comment:
I'm not sure if this was an intentional change, or if you have an auto-formatter that maybe did this, but why did you move this `if row and col then` block outside of the `else` block above? Was this change necessary to make your new picker work?

Reply:
This is intentional. Its not necessary for the picker to work, but is necessary to set the cursor to `row, col`. Otherwise, it just jumps to where the buffer was previously left off, or line 0 of the buffer.

Regardless of whether we want this picker, this change should be merged. For example, lets say we are telescoping the quickfix list, and an open buffer has `N` entries. If we select any of those entries, the cursor wherever the buffer was left previously. With this change, the cursor would be set at the correct `row, col`.

Reply:
I think this fixes: https://github.com/nvim-telescope/telescope.nvim/issues/794

Reply:
oh sweet didn't know there was already an issue for this. Yes I was able to reproduce the issue and fix it.